sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -236,7 +236,26 @@ case class StaticInvoke(
   override def children: Seq[Expression] = arguments
 
   lazy val argClasses = ScalaReflection.expressionJavaClasses(arguments)
-  @transient lazy val method = cls.getDeclaredMethod(functionName, argClasses : _*)
+  @transient lazy val method = {
+    try {
+      cls.getDeclaredMethod(functionName, argClasses: _*)
+    } catch {
+      case _: NoSuchMethodException =>
+        // For some cases, e.g. arg class is Object, `getMethod` cannot find the method.
+        // We look at function name + argument length
+        val m = cls.getDeclaredMethods.filter { m =>
+          m.getName == functionName && m.getParameterCount == arguments.length
+        }
+        if (m.isEmpty) {
+          sys.error(s"Couldn't find $functionName on $cls")
+        } else if (m.length > 1) {
+          // Ambiguous case, we don't know which method to choose, just fail it.
+          sys.error(s"Found ${m.length} $functionName on $cls")
+        } else {
+          m.head
```

Comment:
I get it, but this may not match. I understand that if I want to pass a String, then a method taking an Object "works" but a method taking a Long does not. Does that just cause other problems, or, are we assuming that because it already fails, we just get a different failure, so whatever? you could inspect the args to see if each isAssignableFrom or something I suppose, but maybe not worth it

Reply:
I guess we work best here to find a method for given method name and arguments. For a method taking Object, seems it works to take a String. Adding `isAssignableFrom` check may work for some cases to catch a failure earily. Although as I test, `isAssignableFrom` is unable to catch the case if argument class is int and parameter type is Object too.

Reply:
As I plan to backport this, for further improvement I will leave for master only. This change is only for the missing case. Is any update I should do for this change?

Reply:
I think it's fine to stay conservative for a change you will backport. This will only cause cases that failed completely to possibly work now, right? so I think that's OK.

Reply:
I think it's OK for 2.4.8. It will only cause code that failed outright before to possibly succeed, right? that's OK then as a fix.